RobotNavigation/Maze.cs

```diff
@@ -9,69 +9,64 @@ namespace RobotNavigation
 {
     public class Maze
     {
-        public enum Cell
-        {
-            Empty,
-            Target,
-            Wall
-        }
+        
 
         public int Width { get; set; }
         public int Height { get; set; }
        
-        private Cell[,] _maze;
+        private Position.Cell[,] _maze;
 
         private ArrayList _targets;
-        private Position _pos;
+        private Position _target;
 
         public Maze()
         {
-            _pos = new Position();
+            _target = new Position();
             _targets = new ArrayList();
         }
 
         public Maze(int height, int width)
         {
             Width = width;
             Height = height;
-            init();
+            Init();
         }
 
-        public void init()
+        public void Init()
         {
-            _maze = new Cell[Width, Height];
+            _maze = new Position.Cell[Width, Height];
             for (int i = 0; i < Width; i++)
             {
                 for (int j = 0; j < Height; j++)
                 {
-                    _maze[i, j] = Cell.Empty;
+                    _maze[i, j] = Position.Cell.Empty;
                 }
             }
         }
         
-        public Cell getCell(int x, int y)
+        public Position.Cell GetCell(int x, int y)
         {
             return _maze[x, y];
         }
 
-        public Cell getCell(Maze.Cell cell)
+        public Position.Cell GetCell(Position.Cell cell)
         {
             return cell;
         }
 
-        public void setCell(int x, int y, Cell c)
+        public void SetCell(int x, int y, Position.Cell c)
```

Comment:
This should be SetPosition instead, and should be passed a CellType enum value instead.  See above comment.